src/App/Controller/PublicDiscussionController.php

```diff
@@ -0,0 +1,383 @@
+
+        return $this->render('App:PublicDiscussion:form.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function addProposalAction(Request $request, $slug)
+    {
+        $proposal = $this->getDoctrine()->getRepository('App:Proposal')->findOneBySlug($slug);
+
+        if (null === $proposal) {
+            throw $this->createNotFoundException();
+        }
+
+        $discussion = new PublicDiscussion();
+        $form       = $this->createForm(new EditDiscussionType(), $discussion);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $discussion->setProposal($proposal)->addFollower($this->getUser());
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('public_discussion_show', [
+                'slug' => $discussion->getSlug(),
+            ]));
+        }
+
+        return $this->render('App:PublicDiscussion:form.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_ADMIN')")
+     */
+    public function editAction(Request $request, $slug)
+    {
+        $discussion = $this->getDoctrine()->getRepository('App:PublicDiscussion')->findOneBySlug($slug);
+
+        if (null ===  $discussion) {
+            throw $this->createNotFoundException();
+        }
+
+        $form = $this->createForm(new EditDiscussionType(), $discussion);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('public_discussion_show', [
+                'slug' => $discussion->getSlug(),
+             ]));
+        }
+
+        return $this->render('App:PublicDiscussion:form.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_ADMIN')")
+     */
+    public function deleteAction($slug)
+    {
+        $discussion = $this->getDoctrine()->getRepository('App:PublicDiscussion')->findOneBySlug($slug);
+
+        if (null === $discussion) {
+            throw $this->createNotFoundException();
+        }
+
+        $em = $this->getDoctrine()->getManager();
+
+        if ($discussion->globalDiscussion()) {
+            $em->remove($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('global_discussions'));
+        } elseif ($discussion->themeDiscussion()) {
+            $themeSlug = $discussion->getTheme()->getSlug();
+            $em->remove($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('theme_show', [
+                'slug' => $themeSlug,
+            ]));
+        } else {
```

Comment:
the if, elseif, else is useless here since you return something everytime. You can use 3 if.